numpyro/distributions/distribution.py

```diff
@@ -330,20 +330,16 @@ def __init__(self, base_dist, batch_shape=()):
             batch_shape = self._broadcast_shape(base_dist.batch_shape, batch_shape)
             base_dist = base_dist.base_dist
         self.base_dist = base_dist
-        super().__init__(base_dist.batch_shape, base_dist.event_shape)
-        # adjust batch shape
-        self.expand(batch_shape)
 
-    def expand(self, batch_shape):
+        # adjust batch shape
```

Comment:
@neerajprad I revised the logic here a bit to avoid the fact that `_batch_shape` keeps changing in the constructor. In tensorflow `self._batch_shape`, `self._event_shape` are methods, rather than attributes, so I worried that modifying them will lead to some silent errors.

Reply:
This seems reasonable to me. I think we have tests that cover multiple `expand`s.